python/ray/air/_internal/usage.py

```diff
@@ -0,0 +1,299 @@
+# and entrypoint. At various places in the air/train/tune code base, we may access
+# this data structure as a global variable.
+# Ray client mode makes things a bit tricky - caller should understand that this
+# data structure cannot be accessed across different processes and certainly not
+# between client and head node. See below for env vars to bridge client and head.
+# Mainly for telemetry purposes but may also be used for formatting air console output.
+@dataclass
+class AIRScenario:
+    # One of TRAINER_FIT, TUNER_RESTORE, TUNER_FIT, TUNE_RUN, and TUNE_RUN_EXP.
+    entrypoint = None
+
+    ray_client_mode = None
+
+    trainable = None
+    more_than_one_trial = None
+
+    # only meaningful with "more than one trial" case.
+    search_alg = None
+    scheduler_alg = None
+
+
+# The following three are fields specifically set by ray client and accessed at the
+# head node to set the corresponding field in `AIRScenario`.
+AIR_RAY_CLIENT_MODE_ENV_VAR = "AIR_RAY_CLIENT_MODE"
+AIR_TRAINABLE_ENV_VAR = "AIR_TRAINABLE"
+AIR_ENTRYPOINT_ENV_VAR = "AIR_ENTRYPOINT"
+
+
+_air_scenario: AIRScenario = AIRScenario()
+
+
+def get_air_scenario():
+    return _air_scenario
+
+
+def get_air_scenario_env_vars_to_propagate():
+    # only expected to be called in ray client, right before ray.remote() calls.
+    assert ray.util.client.ray.is_connected()
+    air_scenario_env_vars = {AIR_RAY_CLIENT_MODE_ENV_VAR: "1"}
+
+    assert _air_scenario.entrypoint
+    air_scenario_env_vars.update({AIR_ENTRYPOINT_ENV_VAR: _air_scenario.entrypoint})
+
+    # the original entrypoint is `trainer.fit`.
+    if _air_scenario.entrypoint == TRAINER_FIT:
+        assert _air_scenario.trainable
+        air_scenario_env_vars.update(
+            {AIR_RAY_CLIENT_MODE_ENV_VAR: _air_scenario.trainable}
+        )
+    return air_scenario_env_vars
+
+
+def set_air_scenario_in_head_node():
+    # only expected to be called on head node.
+    # may be called multiple times, but the following times are no-op.
+    # And unset all env vars.
+    assert not ray.util.client.ray.is_connected()
+    if AIR_RAY_CLIENT_MODE_ENV_VAR not in os.environ:  # not coming from ray client
+        assert (
+            AIR_TRAINABLE_ENV_VAR not in os.environ
+            and AIR_ENTRYPOINT_ENV_VAR not in os.environ
+        )
+        return
+    _air_scenario.ray_client_mode = True
+    del os.environ[AIR_RAY_CLIENT_MODE_ENV_VAR]
+    assert AIR_ENTRYPOINT_ENV_VAR in os.environ  # must have entrypoint already set
+    _air_scenario.entrypoint = os.environ.get(AIR_ENTRYPOINT_ENV_VAR)
+    del os.environ[AIR_ENTRYPOINT_ENV_VAR]
+    if AIR_TRAINABLE_ENV_VAR in os.environ:  # trainer.fit + ray client mode
+        assert _air_scenario.entrypoint == TRAINER_FIT
+        _air_scenario.trainable = os.environ.get(AIR_TRAINABLE_ENV_VAR)
+        del os.environ[AIR_TRAINABLE_ENV_VAR]
+    # TODO: how to delete those env vars from runtime_env?
```

Comment:
I'm honestly a bit worried about all this extra logic that needs to be done to support Ray Client, and would like to try find a way to avoid this...